### ISSUE TITLE
 java.lang.IllegalAccessError in io.netty:netty-all:4.1.0.Beta4 on jdk1.6

### ISSUE THREAD

**Issue by junjiemars** [netty/netty#3478]
I'd the following issue (run HttpSnoopClient under jdk1.6):
```java
Exception in thread "main" java.lang.IllegalAccessError: tried to access method java.net.InetSocketAddress.getHostString()Ljava/lang/String; from class io.netty.resolver.DefaultNameResolver
        at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:44)
        at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:30)
        at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:133)
        at io.netty.bootstrap.Bootstrap.doResolveAndConnect(Bootstrap.java:170)
        at io.netty.bootstrap.Bootstrap.connect(Bootstrap.java:138)
        at io.netty.bootstrap.Bootstrap.connect(Bootstrap.java:119)
        at HttpXmlClient.main(HttpXmlClient.java:52)
```

on jdk1.8 it's ok, so what problem? need help.

**Comment by trustin** [comment]
And your Netty version is? I guess it's 4.1, right?

**Comment by trustin** [comment]
IIRC, it's mainly because the default name resolver uses the API that's unavailable in Java 6. `InetSocketAddress.getHostString()`

**Comment by junjiemars** [comment]
Yeath, the nettty version is 4.1.0.Beta4

**Comment by trustin** [comment]
Fixed in 4aa10db9c5d6f5f8bd83f04b6fe9fc979752d92c (master) and 99c4043 (4.1)

**Comment by junjiemars** [comment]
It's awesome.
